Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -366,15 +366,23 @@ extension HTTPClient {
     /// `EventLoopFuture<Response>` of the execution or cancellation of the execution.
     public final class Task<Response> {
         /// `EventLoop` used to execute and process this request.
-        public let eventLoop: EventLoop
-        let promise: EventLoopPromise<Response>
+        public var currentEventLoop: EventLoop {
+            self.lock.withLock {
```

Comment:
you'll need a `return` here, omitting the `return` in properties is a Swift 5.1 feature, sorry

Reply:
Oops, always keep forgetting about this one. Thank you, it is fixed now.